src/test/kotlin/org/taktik/icure/asynclogic/GenericLogicImplTest.kt

```diff
@@ -0,0 +1,57 @@
+package org.taktik.icure.asynclogic
+
+import java.lang.IllegalStateException
+import java.util.UUID
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.flow.count
+import org.junit.jupiter.api.TestInstance
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.taktik.icure.entities.MaintenanceTask
+import org.taktik.icure.test.ICureTestApplication
+
+private const val TEST_CACHE = "build/tests/icureCache"
+
+@SpringBootTest(
+	classes = [ICureTestApplication::class],
+	properties = [
+		"spring.main.allow-bean-definition-overriding=true",
+		"icure.objectstorage.icureCloudUrl=test",
+		"icure.objectstorage.cacheLocation=$TEST_CACHE",
+		"icure.objectstorage.backlogToObjectStorage=true",
+		"icure.objectstorage.sizeLimit=1000",
+		"icure.objectstorage.migrationDelayMs=1000",
+	],
+	webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles("app")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class GenericLogicImplTest (
+	taskLogic: MaintenanceTaskLogic
+): StringSpec({
+
+	"Can create some DTOs if all of them have the rev field equal to none" {
```

Comment:
Problem with this one is that you are creating real maintenance tasks into the database. 
Use mockkClass to create your MaintenanceTaskLogic and mock the behaviour of it. 
```
private val mockedMaintenanceTaskDao = mockk(MaintenanceTaskDAO::class)
every { mockedMaintenanceTaskDao.createEntities(any(), any(), any()) } shouldBe listOf(......)

// Same for other fields

val testedInstance = MaintenanceTaskLogicImpl(mockedMaintenanceTaskDao, mockedSessionLogic, mockedFilters) 
```